In [1]:
import geopandas as gpd
import matplotlib.pyplot as plt

In [5]:
# Reading the shapefile
raw_data_dir = '../raw_data'
newyork_df = gpd.GeoDataFrame.from_file(raw_data_dir + '/NYC/2010_Census_Tracts/geo_export_fef7b0fd-c74c-4d2a-a237-878c8f3d9b90.shp')
print(newyork_df.shape)
newyork_df.head()

(2166, 12)


boro_code boro_ct201      boro_name cdeligibil  ct2010 ctlabel ntacode  \
0         5    5000900  Staten Island          I  000900       9    SI22   
1         1    1009800      Manhattan          I  009800      98    MN19   
2         1    1010000      Manhattan          I  010000     100    MN19   
3         1    1010200      Manhattan          I  010200     102    MN17   
4         1    1010400      Manhattan          I  010400     104    MN17   

                                     ntaname  puma    shape_area   shape_leng  \
0  West New Brighton-New Brighton-St. George  3903  2.497010e+06  7729.016794   
1                    Turtle Bay-East Midtown  3808  1.906016e+06  5534.199811   
2                    Turtle Bay-East Midtown  3808  1.860938e+06  5692.168737   
3                      Midtown-Midtown South  3807  1.860993e+06  5687.802439   
4                      Midtown-Midtown South  3807  1.864600e+06  5693.036367   

                                            geometry  
0  POLYGON ((-74.07920577013245 40.64343078374567...  
1  POLYGON ((-73.96432543478758 40.75638153099091...  
2  POLYGON ((-73.96802436915851 40.75957814005282...  
3  POLYGON ((-73.97124277307127 40.76093641847906...  
4  POLYGON ((-73.97445730550224 40.76229308352487...

There are 2166 census zones.
If we are to represent this as a directed graph (direction of travel matters in traffic), the number of edges we can be calculated using the following formula $n * (n-1)$

$2166 * (2166 - 1)$ 
= $


Network operations may be expensive to 